humility-core/src/hubris.rs

```diff
@@ -2964,7 +2964,7 @@ impl HubrisArchive {
         if let Some(sym) = self.esyms_byname.get("Reset") {
             if let Ok(pc) = core.read_reg(self.arch.as_ref().unwrap().get_pc())
             {
-                if pc >= sym.0 && pc < sym.0 + sym.1 {
+                if (pc as u32) >= sym.0 && (pc as u32) < sym.0 + sym.1 {
```

Comment:
Perhaps a warning if `pc > std::u32::max`?

Reply:
Hmm, that block only runs to check if the target is booted.  I think printing that the PC is out of range would just give a red herring.  I added an additional check before the down casting though.  Now the PC cannot get truncated and falsely trigger the if statement.